metadata_layer.py

```diff
@@ -361,6 +368,46 @@ def __init__(self, content_type, drm_scheme, link=None):
             )
         self.link = link
 
+
+class RecommendationData(object):
+
+    def __init__(self, data_source, identifiers=None):
+        self.data_source = data_source
+        self.identifiers = identifiers or []
```

Comment:
I think you should move this class into model.py, even though there's currently no database model object for recommendations. All the code you've got here deals with the database, and the metadata layer is supposed to be independent of the database. You start here by taking a DataSource object (as opposed to the name of a data source) as an argument, and if that was all it was, I'd just suggest you change that part. But...